src/resolver.cpp

```diff
@@ -132,4 +132,10 @@ namespace libtorrent
 	{
 		m_resolver.cancel();
 	}
+
+	void resolver::set_cache_timeout(seconds timeout)
+	{
+		if (timeout >= seconds(0))
+			m_timeout = timeout;
```

Comment:
where is ``m_timeout`` used? I would have expected a special case of it being zero, where the cache would be circumvented too.